src/blas/backends/rocblas/rocblas_level3.cpp

```diff
@@ -0,0 +1,1365 @@
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        onemkl_rocblas_host_task(cgh, queue, [=](RocblasScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+            auto a_ = reinterpret_cast<const rocDataType_A *>(a);
+            auto b_ = reinterpret_cast<const rocDataType_B *>(b);
+            auto c_ = reinterpret_cast<rocDataType_C *>(c);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, get_rocblas_operation(transa),
+                               get_rocblas_operation(transb), m, n, k, (rocDataType_C *)&alpha, a_,
+                               DT_A, lda, b_, DT_B, ldb, (rocDataType_C *)&beta, c_, DT_C, ldc, c_,
+                               DT_C, ldc, DT_C, rocblas_gemm_algo_standard, 0,
+                               rocblas_gemm_flags_pack_int8x4);
+        });
+    });
+    return done;
+}
+
+#define GEMM_EX_LAUNCHER_USM(TYPE_A, TYPE_B, TYPE_C, ROCBLAS_ROUTINE, CUDADATATYPE_A,            \
+                             CUDADATATYPE_B, CUDADATATYPE_C)                                     \
+    cl::sycl::event gemm(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,  \
+                         int64_t n, int64_t k, TYPE_C alpha, const TYPE_A *a, int64_t lda,       \
+                         const TYPE_B *b, int64_t ldb, TYPE_C beta, TYPE_C *c, int64_t ldc,      \
+                         const cl::sycl::vector_class<cl::sycl::event> &dependencies) {          \
+        return gemm(ROCBLAS_ROUTINE, CUDADATATYPE_A, CUDADATATYPE_B, CUDADATATYPE_C, queue,      \
+                    transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc, dependencies); \
+    }
+
+GEMM_EX_LAUNCHER_USM(sycl::half, sycl::half, float, rocblas_gemm_ex, rocblas_datatype_f16_r,
+                     rocblas_datatype_f16_r, rocblas_datatype_f32_r)
+GEMM_EX_LAUNCHER_USM(sycl::half, sycl::half, sycl::half, rocblas_gemm_ex, rocblas_datatype_f16_r,
+                     rocblas_datatype_f16_r, rocblas_datatype_f16_r)
+
+#undef GEMM_EX_LAUNCHER_USM
+
+cl::sycl::event gemm(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,
+                     int64_t n, int64_t k, float alpha, const bfloat16 *a, int64_t lda,
+                     const bfloat16 *b, int64_t ldb, float beta, float *c, int64_t ldc,
+                     const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    throw unimplemented("blas", "gemm", "for column_major layout");
+}
+template <typename Func, typename T>
+inline cl::sycl::event symm(Func func, cl::sycl::queue &queue, side left_right, uplo upper_lower,
+                            int64_t m, int64_t n, T alpha, const T *a, int64_t lda, const T *b,
+                            int64_t ldb, T beta, T *c, int64_t ldc,
+                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
```

Comment:
Here and other locations:
```suggestion
                            const std::vector<cl::sycl::event> &dependencies) {
```